clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2200,7 +2197,7 @@ static void printTemplateArguments(ASTContext &Ctx, raw_ostream &ArgOS,
 static std::string getKernelNameTypeString(QualType T, ASTContext &Ctx,
                                            const PrintingPolicy &TypePolicy) {
 
-  QualType FullyQualifiedType = TypeName::getFullyQualifiedType(T, Ctx, true);
+  QualType FullyQualifiedType = T.getCanonicalType();
```

Comment:
So I cannot help but wonder if this variable is even necessary anymore...

Reply:
I have changed that.